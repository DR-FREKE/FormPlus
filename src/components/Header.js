import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import styles from "../styles/Header.module.css";
import { IoIosSearch } from "react-icons/io";

export const Dropdown = ({ name, children, ...props }) => (
  <fieldset>
    <legend>{name}</legend>
    <select name={name} value={props.value} onChange={(e) => props.setValue(e)}>
      {children}
    </select>
  </fieldset>
);

const Header = (props) => {
  const [category, setCategory] = useState("");
  const [order, setOrder] = useState("");
  const [date, setDate] = useState("");
  const [search, setSearch] = useState("");

  useEffect(() => {
    if (category != "") props.runFilter(category);
    if (order != "") props.sortOrder(order);
    if (date != "") props.sortDate(date);
  }, [category, order, search, date]);

  const handleCategoryChange = (e) => {
    setCategory(e.target.value);
    setOrder("");
    setDate("");
    setSearch("");
  };

  const handleOrderChange = (e) => {
    setOrder(e.target.value);
  };

  const handleDateChange = (e) => {
    setDate(e.target.value);
  };

  const handleSearchChange = (e) => {
    setSearch(e.target.value, checkSearchField(e.target.value));
  };

  const checkSearchField = (val) => {
    props.search(val);
  };

  return (
    <div className={styles.topBar}>
      <div className={styles.searchDiv}>
        <div className={styles.searchInput}>
          <input
            placeholder="Search Template"
            name="search"
            value={search}
            onChange={handleSearchChange}
          />
          <IoIosSearch
            style={{ cursor: "pointer", fontSize: "20px" }}
            onClick={() => props.search(search)}
          />
        </div>
      </div>
      <div className={styles.sortDiv}>
        <span>Sort By:</span>
        <Dropdown
          name="category"
          value={category}
          setValue={handleCategoryChange}>
          <option value="All Templates">All</option>
          <option value="Education">Education</option>
          <option value="E-commerce">E-commerce</option>
          <option value="Health">Health</option>
        </Dropdown>
        <Dropdown name="order" value={order} setValue={handleOrderChange}>
          <option value="default">Default</option>
          <option value="asc">Ascending</option>
          <option value="dsc">Desending</option>
        </Dropdown>
        <Dropdown name="date" value={date} setValue={handleDateChange}>
          <option value="default">Default</option>
          <option value="asc">Ascending</option>
          <option value="dsc">Desending</option>
        </Dropdown>
      </div>
    </div>
  );
};

export default Header;
