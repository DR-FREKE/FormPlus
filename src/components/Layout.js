import React from "react";
import PropTypes from "prop-types";
import styles from "../styles/Layout.module.css";

const Layout = ({ children }) => (
  <>
    <div className={styles.container}>
      <main className={styles.main}>{children}</main>
    </div>
  </>
);

Layout.propTypes = {};

export default Layout;
