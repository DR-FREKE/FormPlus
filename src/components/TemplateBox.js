import React from "react";
import styles from "../styles/Template.module.css";

const TemplateBox = (props) => (
  <div className={styles.box}>
    <div className={styles.topBox}>
      <div className={styles.topHeader}>
        <span>{props.template_name}</span>
      </div>
      <div className={styles.description}>
        <span>{props.description}</span>
        <div className={styles.category}>
          <span>Categories {JSON.stringify(props.categories)}</span>
        </div>
      </div>
    </div>
    <div className={styles.footerBox}>
      <span>use template</span>
    </div>
  </div>
);

export default TemplateBox;
