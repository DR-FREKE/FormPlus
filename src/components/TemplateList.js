import React from "react";
import { connect } from "react-redux";
import styles from "../styles/Template.module.css";
import TemplateBox from "../components/TemplateBox";

const TemplateList = ({ templates, total, name, ...props }) => (
  <div className={styles.mainContainer}>
    <div className={styles.headerTitle}>
      <span>{name || "All Templates"}</span>
      <span>
        {total} templates{" "}
        {props.filtering == true && name != "All Templates" ? "found" : ""}
      </span>
    </div>
    <div className={styles.grid}>
      {templates && templates.map((content) => <TemplateBox {...content} />)}
    </div>
  </div>
);

const mapStateToProps = (state) => ({
  filtering: state.template.filtering,
  sorting: state.template.sorting,
  date_sorting: state.template.date_sorting,
  searching: state.template.searching,
});

export default connect(mapStateToProps, {})(TemplateList);
