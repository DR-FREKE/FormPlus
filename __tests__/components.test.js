import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import Header from "../src/components/Header";
import Layout from "../src/components/Layout";
import TemplateBox from "../src/components/TemplateBox";
import TemplateList from "../src/components/TemplateList";

const mockStore = configureStore([]);

/** test for Layout of the page */
describe("test for AppLayout", () => {
  const layout = renderer.create(<Layout />);

  it("should render the Layout component", () => {
    expect(layout.toJSON()).toMatchSnapshot();
  });

  it("should have children", () => {
    const child_length = layout.toJSON().children.length;
    expect(child_length).toBeGreaterThan(0);
  });
});

/** test for the top header */
describe("test for header component", () => {
  const header = renderer.create(<Header />);

  it("should render the Header component", () => {
    expect(header.toJSON()).toMatchSnapshot();
  });

  it("should test input is rendered", () => {
    const input_field = header.root;
    expect(input_field.findByType("input").type).toMatchSnapshot();
  });
});

/** test for template listing */
describe("test for template listing", () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      template: {
        filtering: false,
      },
    });
  });
  it("should render template box", () => {
    const template_box = renderer.create(<TemplateBox />).toJSON();
    expect(template_box).toMatchSnapshot();
  });

  it("should render all boxes as list", () => {
    const template_list = renderer
      .create(
        <Provider store={store}>
          <TemplateList />
        </Provider>
      )
      .toJSON();
    expect(template_list).toMatchSnapshot();
  });
});
