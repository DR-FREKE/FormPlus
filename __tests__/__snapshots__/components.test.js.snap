// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for AppLayout should render the Layout component 1`] = `
<div
  className="container"
>
  <main
    className="main"
  />
</div>
`;

exports[`test for header component should render the Header component 1`] = `
<div
  className="topBar"
>
  <div
    className="searchDiv"
  >
    <div
      className="searchInput"
    >
      <input
        placeholder="Search Template"
      />
      <i
        className="fa fa-search"
        style={
          Object {
            "fontWeight": "lighter",
          }
        }
      />
    </div>
  </div>
  <div
    className="sortDiv"
  >
    <span>
      Sort By:
    </span>
    <fieldset>
      <legend>
        Category
      </legend>
      <select
        name="some name"
        onChange={[Function]}
        value=""
      >
        <option
          value="All"
        >
          All
        </option>
        <option
          value="Education"
        >
          Education
        </option>
        <option
          value="E-commerce"
        >
          E-commerce
        </option>
        <option
          value="Health"
        >
          Health
        </option>
      </select>
    </fieldset>
    <fieldset>
      <legend>
        Order
      </legend>
      <select
        name="some name"
        onChange={[Function]}
        value=""
      >
        <option>
          Default
        </option>
      </select>
    </fieldset>
    <fieldset>
      <legend>
        Date
      </legend>
      <select
        name="some name"
        onChange={[Function]}
        value=""
      >
        <option>
          Default
        </option>
      </select>
    </fieldset>
  </div>
</div>
`;

exports[`test for header component should test input is rendered 1`] = `"input"`;

exports[`test for template listing should render all boxes as list 1`] = `
<div
  className="mainContainer"
>
  <div
    className="headerTitle"
  >
    <span>
      All Templates
    </span>
    <span>
       templates
    </span>
  </div>
  <div
    className="grid"
  />
</div>
`;

exports[`test for template listing should render template box 1`] = `
<div
  className="box"
>
  <div
    className="topBox"
  >
    <div
      className="topHeader"
    >
      <span />
    </div>
    <div
      className="description"
    >
      <span />
    </div>
  </div>
  <div
    className="footerBox"
  >
    <span>
      use template
    </span>
  </div>
</div>
`;
